### PR TITLE
add e2e test for issue 9874

### LINKED ISSUE #9874: An error occurred when daemonset modified Scaling and Upgrade Policy
**Setup**
- Rancher version: v2.7.8
- Rancher UI Extensions:
- Browser type & version: chrome 117.0.5938.149

**Describe the bug**

An error occurred when daemonset modified Scaling and Upgrade Policy
**To Reproduce**

1. Navigate to daemonset and modify Scaling and Upgrade Policy to `On Delete`
2. Save

**Result**
Unable to save with error message：
```
DaemonSet.apps "demo" is invalid: spec.updateStrategy: Unsupported value: apps.DaemonSetUpdateStrategy{Type:"Delete", RollingUpdate:(*apps.RollingUpdateDaemonSet)(nil)}: supported values: "RollingUpdate", "OnDelete"
```

**Expected Result**

**Screenshots**

image

**Additional context**

### PR DESCRIPTION
**Pull request by aalves08**
### Summary
Fixes #9874 (e2e testing)

### Occurred changes and/or fixed issues

- Adds pattern to e2e tests for editing a `daemonset` on a fake cluster
- Add e2e test to prevent regression of issue #9874 , related to Scaling and Upgrade Policy sending the wrong type string for option `On Delete`

### Checklist
- [ ] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [ ] The PR has a Milestone 
- [ ] The PR template has been filled out
- [ ] The PR has been self reviewed 
- [ ] The PR has a reviewer assigned
- [ ] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [ ] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: cypress/e2e/tests/pages/explorer/workloads/daemonsets.spec.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    describe('Daemonsets', () => {
+      it('modifying "Scaling and Upgrade Policy" to "On Delete" should use the correct property "OnDelete"', () => {
+        // same name as in blueprints/nav/k8s-schemas/apps.daemonsets
+        const daemonsetName = 'calico-node';
+
+        // list view daemonsets
+        const workloadsDaemonsetsListPage = new WorkloadsDaemonsetsListPagePo(fakeMgmtClusterId);
+
+        workloadsDaemonsetsListPage.goTo();
+        workloadsDaemonsetsListPage.listElementWithName(daemonsetName).should('exist');
```
Comment (momesgin) [inline cypress/e2e/tests/pages/explorer/workloads/daemonsets.spec.ts:39]: I think we would prefer asserting the visibility instead

File: cypress/e2e/tests/pages/explorer/workloads/daemonsets.spec.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        // edit view daemonsets
+        const workloadsDaemonsetsEditPage = new WorkLoadsDaemonsetsEditPagePo(fakeMgmtClusterId);
+
+        workloadsDaemonsetsEditPage.clickTab('#DaemonSet');
+        workloadsDaemonsetsEditPage.clickTab('#upgrading');
+        workloadsDaemonsetsEditPage.ScalingUpgradePolicyRadioBtn().set(1);
+        workloadsDaemonsetsEditPage.saveCreateForm().click();
+
+        workloadsDaemonsetsListPage.listElementWithName(daemonsetName).should('exist');
```
Comment (momesgin) [inline cypress/e2e/tests/pages/explorer/workloads/daemonsets.spec.ts:50]: same here

File: cypress/e2e/tests/pages/explorer/workloads/daemonsets.spec.ts (hunk 1 above)
Comment (aalves08) [reply cypress/e2e/tests/pages/explorer/workloads/daemonsets.spec.ts]: change done. I verified that works, but fundamentally `exists` is different than `visible`... Imagine that the daemonset row exists but is not visible in the table (needs scrolling). That doesn't mean it doesn't exist on the list... It should be clickable by the user. 

The `visible` assertion would fail on that particular case.

File: cypress/e2e/tests/pages/explorer/workloads/daemonsets.spec.ts (hunk 2 above)
Comment (aalves08) [reply cypress/e2e/tests/pages/explorer/workloads/daemonsets.spec.ts]: same comment as before

File: cypress/e2e/tests/pages/explorer/workloads/daemonsets.spec.ts (hunk 1 above)
Comment (momesgin) [reply cypress/e2e/tests/pages/explorer/workloads/daemonsets.spec.ts]: @aalves08 that's a very good point, my concern with `exists` is that even a hidden element with zero width and height can still pass the test, but as you mentioned there are some edge cases as well.

**Review by momesgin — APPROVED** [review]
lgtm
